src/Bicep.Deploy/DeploymentManager.cs

```diff
@@ -19,27 +20,69 @@ public async Task<ArmDeploymentResource> CreateOrUpdateAsync(ArmDeploymentDefini
+                    var subscription = await armClient.GetDefaultSubscriptionAsync(cancellationToken);
+                    deploymentDefinition = deploymentDefinition with { SubscriptionId = subscription.Id };
+                }
 
-            var deploymentResource = armClient.GetArmDeploymentResource(deploymentDefinition.Id);
-            var deploymentContent = new ArmDeploymentContent(deploymentDefinition.Properties);
+                var deploymentResource = armClient.GetArmDeploymentResource(deploymentDefinition.Id);
+                var deploymentContent = new ArmDeploymentContent(deploymentDefinition.Properties);
 
-            var operation = await deploymentResource.UpdateAsync(WaitUntil.Completed, deploymentContent, cancellationToken);
+                var operation = await deploymentResource.UpdateAsync(WaitUntil.Completed, deploymentContent, cancellationToken);
 
-            return operation.Value;
+                return operation.Value;
+            }
+            catch (RequestFailedException ex)
+            {
+                throw new DeploymentException(ex.Message, ex);
+            }
         }
 
-        public Task<ArmDeploymentResource> CreateOrUpdateAsync(ArmDeploymentDefinition deploymentDefinition, Action<ImmutableSortedSet<ArmDeploymentOperation>>? onOperationsUpdated, CancellationToken cancellationToken)
+        public async Task<ArmDeploymentResource> CreateOrUpdateAsync(ArmDeploymentDefinition deploymentDefinition, Action<ImmutableSortedSet<ArmDeploymentOperation>>? onOperationsUpdated, CancellationToken cancellationToken)
         {
             // TODO: Custom polling logic. Take a look at:
             // - https://github.com/Azure/azure-powershell/blob/b819adaeec70e735e3b46d51c72bc2fc17c4fef6/src/Resources/ResourceManager/SdkClient/NewResourceManagerSdkClient.cs#L201
             // - Azure.Core.OperationPoller
-            throw new NotImplementedException();
+            //throw new NotImplementedException();
+            var armClient = new ArmClient(new DefaultAzureCredential());
+
+            try
+            {
+                if (deploymentDefinition.SubscriptionId is { } subscriptionId &&
+                    subscriptionId.Equals(Guid.Empty.ToString(), StringComparison.OrdinalIgnoreCase))
+                {
+                    var subscription = await armClient.GetDefaultSubscriptionAsync(cancellationToken);
+                    deploymentDefinition = deploymentDefinition with { SubscriptionId = subscription.Id };
+                }
+
+                var deploymentResource = armClient.GetArmDeploymentResource(deploymentDefinition.Id);
+                var deploymentContent = new ArmDeploymentContent(deploymentDefinition.Properties);
+
+                var operation = await deploymentResource.UpdateAsync(WaitUntil.Started, deploymentContent, cancellationToken);
+
+                var currentOperations = new List<ArmDeploymentOperation>();
+                while (!operation.HasCompleted)
```

Comment:
We can simplify pagination handling by using [System.Linq.Async](https://www.nuget.org/packages/System.Linq.Async). Here's an example from the Azure SDK documentation: https://learn.microsoft.com/en-us/dotnet/azure/sdk/pagination#use-systemlinqasync-with-asyncpageable